backend/kale/common/podutils.py

```diff
@@ -300,3 +307,63 @@ def compute_component_id(pod):
     component_id = component_name + "@sha256=" + component_spec_digest
     log.info("Computed component ID: %s", component_id)
     return component_id
+
+
+def get_snapshotclasses(label_selector=""):
+    """List snapshotclasses."""
+    co_client = k8sutils.get_co_client()
+
+    snapshotclasses = co_client.list_cluster_custom_object(
+        group="snapshot.storage.k8s.io",
+        version="v1beta1",
+        plural="volumesnapshotclasses",
+        label_selector=label_selector)
+    return snapshotclasses
+
+
+def list_snapshotclass_storage_provisioners(label_selector=""):
+    """List the storage provisioners of the snapshotclasses."""
+    snapshotclasses = get_snapshotclasses(label_selector)["items"]
+    snapshotclass_provisioners = []
+    for i in snapshotclasses:
+        snapshotclass_provisioners.append(i["driver"])
+    return snapshotclass_provisioners
+
+
+def check_snapshot_availability():
+    """Check if snapshotclasses are available for notebook."""
+    client = k8sutils.get_v1_client()
+    namespace = get_namespace()
+    pod_name = get_pod_name()
+    pod = client.read_namespaced_pod(pod_name, namespace)
+    snapshotclass_provisioners = list_snapshotclass_storage_provisioners()
+
+    for volume in pod.spec.volumes:
+        pvc = volume.persistent_volume_claim
+        if not pvc:
+            continue
+        pvc = client.read_namespaced_persistent_volume_claim(pvc.claim_name,
```

Comment:
Don't re-use the same variable name. Either name the first `pvc` differently, or this one

Reply:
I didn't change this from how it already was in Kale. https://github.com/kubeflow-kale/kale/blob/07ccdc5bdc4c39a3c04dad44d4c07447378efb98/backend/kale/common/podutils.py#L151

Reply:
I also found this a strange way of doing things, but didn't think the fix should be included in the addition of these features.

Reply:
Yes you are referring to the `_list_volumes` function. Indeed, I will fix this in the original function as well